app/services/ipam/random_db.rb

```diff
@@ -0,0 +1,37 @@
+module IPAM
+  # Internal DB IPAM returning all IPs in random order to minimize race conditions
+  class RandomDb < Base
+    attr_accessor :generator
+
+    def initialize(opts = {})
+      super(opts)
+      self.generator ||= Random.new(mac ? mac.gsub(':', '').to_i(16) : Random.new_seed)
```

Comment:
@lzap: Sorry, one more thing. You don't need to overwrite the `initialize` method. Just define a new method `generator`:

```
  def generator
     @generator ||= Random.new(mac ? mac.gsub(':', '').to_i(16) : Random.new_seed)
  end
```